cms/static/js/spec/views/course_video_settings_spec.js

```diff
@@ -464,6 +464,30 @@ define(
                 verifyOrganizationCredentialField('api-key', 'API Key');
             });
 
+            it('shows username input field if selected provider is Cielo', function() {
+                // Cielo is currently selected provider.
+                // Click change button to render organization credentials view.
+                $courseVideoSettingsEl.find('.action-change-provider').click();
+
+                // Verify cielo api key and username fields are present.
+                verifyOrganizationCredentialField('api-key', 'API Key');
+                verifyOrganizationCredentialField('username', 'Username');
+            });
+
+            it('does not show username input field if selected provider is 3Play Media', function() {
+                // Set selected provider to 3Play Media
+                changeProvider('3PlayMedia');
+
+                // Verify cielo username field is not present.
+                expect(
+                    $courseVideoSettingsEl.find('.' + courseVideoSettingsView.selectedProvider + '-username')
+                ).not.toExist();
+
+                // Verify api key and api secret input fields are present.
+                verifyOrganizationCredentialField('api-key', 'API Key');
+                verifyOrganizationCredentialField('api-secret', 'API Secret');
+            });
```

Comment:
As discussed we can write a helper that should verify

* length of input fields to be rendered
* selector and text of each individual field